### PR TITLE
[FLINK-16155][docs] Translate "Operator/Process Function" into Chinese

### PR DESCRIPTION
**Pull request by liboshuai01**
What is the purpose of the change
---
Translate Process Function page into Chinese

Verifying this change
---
This change is a trivial rework / code cleanup without any test coverage.

Does this pull request potentially affect one of the following parts:
---
- Dependencies (does it add or upgrade a dependency): (no)
- The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
- The serializers: (yes / no / don't know)
- The runtime per-record code paths (performance sensitive): (no)
- Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
- The S3 file system connector: (no)

Documentation
---
- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by liboshuai01** [comment]
Hi, @leonardBang @fredia , Excuse me, do you have time to take a look at this?

File: docs/content.zh/docs/dev/datastream/operators/process_function.md [hunk 1]
```diff
@@ -26,68 +26,61 @@ under the License.
 
 # Process Function
 
-## The ProcessFunction
+## ProcessFunction 简介
 
-The `ProcessFunction` is a low-level stream processing operation, giving access to the basic building blocks of
-all (acyclic) streaming applications:
+`ProcessFunction` 是一种低级别的流处理操作，基于它用户可以访问（无环）流应用程序的所有基本构建块：
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/operators/process_function.md:31]: 文中多次出现的 `low-level` 一词是不是可以翻译成`底层`?

File: docs/content.zh/docs/dev/datastream/operators/process_function.md [hunk 2]
```diff
@@ -26,68 +26,61 @@ under the License.
...
+  - 从某个输入接收元素时更新状态。
+  - 从另一个输入接收元素时，查询状态并生成 join 结果。
 
-For example, you might be joining customer data to financial trades,
-while keeping state for the customer data. If you care about having
-complete and deterministic joins in the face of out-of-order events,
-you can use a timer to evaluate and emit the join for a trade when the
-watermark for the customer data stream has passed the time of that
-trade.
+例如，你可能会将客户数据与金融交易进行 join，同时想要保留客户数据的状态。如果你希望即使在出现乱序事件时仍然可以得到完整且确定的 join 结果，你可以通过注册一个定时器在客户数据流的 watermark 已经超过当前这条 finance trade 记录时计算和发送 join 结果。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/operators/process_function.md:69]: `已经超过当前这条 finance trade 记录时` -> `已经超过当前这条金融交易记录时`

前文的 `finance trade` 也被翻译了，可以尽量保持一致。

File: docs/content.zh/docs/dev/datastream/operators/process_function.md [hunk 3]
```diff
@@ -26,68 +26,61 @@ under the License.
...
 
-## Example
+## 示例
 
-In the following example a `KeyedProcessFunction` maintains counts per key, and emits a key/count pair whenever a minute passes (in event time) without an update for that key:
+在下面的例子中，`KeyedProcessFunction` 维护每个键的计数，并且每次超过一分钟（事件时间）没有更新时输出一次（键，计数）对。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/operators/process_function.md:73]: `（键，计数）对` -> `键/计数对`

使用括号会不会有些像注解？下文相同

File: docs/content.zh/docs/dev/datastream/operators/process_function.md [hunk 4]
```diff
@@ -26,68 +26,61 @@ under the License.
 
 # Process Function
 
-## The ProcessFunction
+## ProcessFunction 简介
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/operators/process_function.md:29]:
```suggestion
## ProcessFunction
```

File: docs/content.zh/docs/dev/datastream/operators/process_function.md [hunk 5]
```diff
@@ -338,17 +331,14 @@ if __name__ == '__main__':
...
-harmful because processing-time timestamps are indeterministic and not aligned with watermarks. Besides, user-implemented logic
-depends on this wrong timestamp highly likely is unintendedly faulty. So we've decided to fix it. Upon upgrading to 1.4.0, Flink jobs
-that are using this incorrect event-time timestamp will fail, and users should adapt their jobs to the correct logic.
+在 Flink 1.4.0 之前，在调用处理时间定时器时，`ProcessFunction.onTimer()` 方法将当前的处理时间设置为事件时间的时间戳。此行为非常不明显，用户可能不会注意到。
+然而，这样做是有害的，因为处理时间的时间戳是不确定的，并且和 watermark 不一致。此外，用户依赖于此错误的时间戳来实现逻辑很有可能导致非预期的错误。
+因此，我们决定对其进行修复。在 1.4.0 后，使用此错误的事件时间时间戳的 Flink 作业将失败，用户应将其作业更正为正确的逻辑。
 {{< /hint >}}
 
-## The KeyedProcessFunction
+## KeyedProcessFunction 简介
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/operators/process_function.md:339]:
```suggestion
## KeyedProcessFunction
```

**Review by PatrickRen — COMMENTED** [review]
@liboshuai01 Thanks for the awesome translation! I left some comments.

File: docs/content.zh/docs/dev/datastream/operators/process_function.md (hunk 1 above)
Comment (liboshuai01) [reply docs/content.zh/docs/dev/datastream/operators/process_function.md]: `底层`确实更为适合，已采纳修改。

File: docs/content.zh/docs/dev/datastream/operators/process_function.md (hunk 2 above)
Comment (liboshuai01) [reply docs/content.zh/docs/dev/datastream/operators/process_function.md]: 是的，已采纳修改。

File: docs/content.zh/docs/dev/datastream/operators/process_function.md (hunk 3 above)
Comment (liboshuai01) [reply docs/content.zh/docs/dev/datastream/operators/process_function.md]: 已采纳，并修改。

File: docs/content.zh/docs/dev/datastream/operators/process_function.md (hunk 4 above)
Comment (liboshuai01) [reply docs/content.zh/docs/dev/datastream/operators/process_function.md]: 已采纳，并修改。

File: docs/content.zh/docs/dev/datastream/operators/process_function.md (hunk 5 above)
Comment (liboshuai01) [reply docs/content.zh/docs/dev/datastream/operators/process_function.md]: 已采纳，并修改。

**Comment by liboshuai01** [comment]
@PatrickRen Excuse me, do you have time to take a look at this?

**Review by PatrickRen — APPROVED** [review]
@liboshuai01 Thanks for the update! LGTM

**Pull request merged**